streams_map_outgoing.go

```diff
@@ -60,7 +60,7 @@ func (m *outgoingStreamsMap[T]) OpenStream() (T, error) {
 	// if there are OpenStreamSync calls waiting, return an error here
 	if len(m.openQueue) > 0 || m.nextStream > m.maxStream {
 		m.maybeSendBlockedFrame()
-		return *new(T), streamOpenErr{errTooManyOpenStreams}
+		return *new(T), streamOpenErr{&StreamLimitReachedError{}}
```

Comment:
This should be 
`streamOpenErr{StreamLimitReachedError{}} // no pointer `

Otherwise `err.Temporary` wont be true.

Reply:
Or alternatively, the `errors.Is` test should test for the pointer.